src/services/importTracker.ts

```diff
@@ -140,6 +146,20 @@ namespace ts.FindAllReferences {
             }
         }
 
+        function handleImportCall(importCall: ImportCall) {
+            const top = findAncestor(importCall, isAmbientModuleDeclaration) || importCall.getSourceFile();
+            const exported = isExported(importCall, top);
+            if (exported) addIndirectUsers(top);
+            else addIndirectUser(top);

```

Comment:
I'd recommend refactoring `addIndirectUser`(s) as follows:

1. inline `addIndirectUser` to the top of `addIndirectUsers` and rename the latter to `addIndirectUser`.
2. add a new parameter to `addIndirectUsers`: `addTransitiveDependencies = false`.
2. Change the original first line of `addIndirectUsers` to `if (!isNew || !addTransitiveDependencies) {`
3. In two of the existing uses of `addIndirectUsers`, hard-code the new argument to `true`.
4. In this use, and the other one like it, pass `exported` as the second argument. (`isReexport ...` for the other use